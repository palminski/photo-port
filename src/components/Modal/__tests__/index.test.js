import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '../';

const mockToggleModal = jest.fn();

const currentPhoto = {
    name: 'Park Bench',
    category: 'Landscape',
    description: 'Test',
    index: 1
};


afterEach(cleanup);

describe('Modal Component', () => {
    it('renders', () => {
        render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal}/>);
    });

    it('matches snap', () => {
        const {asFragment} = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal}/>);
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Click Event', () => {
    it('calls onClose Handler', () => {
        const {getByText} = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal}/>);
        fireEvent.click(getByText('Close this modal.'));
        expect(mockToggleModal).toBeCalledTimes(1);
    });
})


