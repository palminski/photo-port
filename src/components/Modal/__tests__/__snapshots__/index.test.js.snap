// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snap 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park Bench
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Test
      </p>
      <button
        type="button"
      >
        Close this modal.
      </button>
    </div>
  </div>
</DocumentFragment>
`;
